### PR TITLE
reference: update the description about the `tidb_enable_table_partition` variable

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What is changed, added or deleted? (Required)

Update the description about the `tidb_enable_table_partition` variable.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:  https://github.com/pingcap/docs-cn/pull/2718
- Other reference link(s):

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp

**Comment by yikeke** [comment]
Please help follow up this PR, thx~ @ran-huang

**Comment by tiancaiamao** [comment]
Done @yikeke @jackysp

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -276,12 +276,13 @@ set @@global.tidb_distsql_scan_concurrency = 10
...
 
-- Scope: SESSION
-- Default value: "auto"
+- Scope: SESSION | GLOBAL
+- Default value: "on"
 - This variable is used to set whether to enable the `TABLE PARTITION` feature.
     - `off` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.
-    - `auto` indicates enabling the `TABLE PARTITION` feature. Currently, it indicates enabling range partition and hash partition.
-    - `on` functions the same way as `auto` does.
+    - `on` indicates enabling the `TABLE PARTITION` feature for the supported partition type. Currently, it indicates enabling range partition and hash partition and range columns partition with the single column case.
```
Comment (yikeke) [inline reference/configuration/tidb-server/tidb-specific-variables.md:283]:
```suggestion
    - `on` indicates enabling the `TABLE PARTITION` feature for the supported partition types. Currently, it indicates enabling range partition, hash partition and range column partition with one single column.
```

File: reference/configuration/tidb-server/tidb-specific-variables.md [hunk 2]
```diff
@@ -276,12 +276,13 @@ set @@global.tidb_distsql_scan_concurrency = 10
...
+    - `auto` functions the same way as `on` does.
```
Comment (yikeke) [inline reference/configuration/tidb-server/tidb-specific-variables.md:284]:
```suggestion
    - Currently, `auto` has the same function as `on` does.
```

**Pull request merged**